src/fides/api/schemas/language.py

```diff
@@ -0,0 +1,46 @@
+from enum import Enum
+from os.path import dirname, join
+from typing import Dict
+
+import yaml
+from pydantic import BaseModel
+
+from fides.api.util.text import to_snake_case
+from fides.config.helpers import load_file
+
+LANGUAGES_YAML_FILE_PATH = join(
+    dirname(__file__),
+    "../../data",
+    "language",
+    "languages.yml",
+)
+
+
+class Language(BaseModel):
+    id: str
+    name: str
+
+
+def _load_supported_languages() -> Dict[str, Language]:
+    """Loads language dict based on yml file on disk"""
+    with open(load_file([LANGUAGES_YAML_FILE_PATH]), "r", encoding="utf-8") as file:
+        _languages = yaml.safe_load(file).get("languages", [])
+        language_dict = {}
+        for language in _languages:
+            language_dict[language["id"]] = Language.parse_obj(language)
+        return language_dict
+
+
+supported_languages_by_id: Dict[str, Language] = (
+    _load_supported_languages()
+)  # should only be accessed for read-only access
+
+
+# dynamically create an enum based on definitions loaded from YAML
+SupportedLanguage = Enum(  # type: ignore[misc]
+    "SupportedLanguage",
+    {
+        to_snake_case(language.name): language.id
```

Comment:
What language format will we add to the translations then, "portuguese_brazil"?

Reply:
hmm no the way i imagine this working is that the _value_ of the enum (i.e. the Language `id`) is what would actually be stored on the translation resource -- so it'd be `pt-BR`. the `test_language_enum_in_schema` shows an example.

does this make sense to you? i may be doing something different than what you were imagining!

Reply:
it can just be a little painful when the keys and values don't match, transforming back and forth from schemas to databases to dictionaries etc, we'll just need to track down all the places that need attending to!

```python
LookupError: 'sr-Cyrl' is not among the defined enum values. Enum name: language. Possible values: en_us, en_gb, portuguese_.., serbi...
```

```python
E           pydantic.error_wrappers.ValidationError: 1 validation error for ExperienceConfigCreateTemplate
E           translations -> 0 -> language
E             value is not a valid enumeration member; permitted: 'en_us', 'en_gb', 'pt-BR', 'sr-Cyrl' (type=type_error.enum; enum_values=[<Language.en_us: 'en_us'>, <Language.en_gb: 'en_gb'>, <Language.portuguese_brazil: 'pt-BR'>, <Language.serbian_cyrillic: 'sr-Cyrl'>])
```

Reply:
got it, agreed. we discussed more offline - we _may_ look to eventually align keys and values to match, but we'll merge as-is for now, and @pattisdr will give a try integrating with her ongoing work to see how painful it is. so we may revisit this!